# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build freetype-build sdl-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --print-requires sdl-build | grep -q egl-opengl-stubs-build && GL=1

# configure (ac-2.61; am-1.10; lt-2.2.6)
libtoolize -c -f; aclocal -I acinclude; autoconf; automake -a -c
PKG_CONFIG_PATH=/x11/lib/pkgconfig LDFLAGS=-Wl,-rpath,/x11/lib ./configure --disable-static --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/bin
test $GL && install .libs/glfont $DESTDIR/x11/bin
install .libs/showfont $DESTDIR/x11/bin
install -d $DESTDIR/x11/include/SDL
install -m 644 SDL_ttf.h $DESTDIR/x11/include/SDL
install -d $DESTDIR/x11/lib
install .libs/libSDL_ttf-1.2.so.0.* $DESTDIR/x11/lib/libSDL_ttf-1.2.so.0
ln -sf libSDL_ttf-1.2.so.0 $DESTDIR/x11/lib/libSDL_ttf.so
install -d $DESTDIR/x11/lib/pkgconfig
install -m 644 SDL_ttf.pc $DESTDIR/x11/lib/pkgconfig

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/sdl_ttf-build.pc << EOF
Name: SDL_ttf
Version: 2.0.11
Description: TrueType font library for SDL
Requires: $REQUIRES

devel=\\
/x11/include/SDL/SDL_ttf.h \\
/x11/lib/libSDL_ttf.so \\
/x11/lib/pkgconfig/SDL_ttf.pc

exec=\\
EOF
test $GL && cat >> $DESTDIR/x11/share/pkgconfig/sdl_ttf-build.pc << EOF
/x11/bin/glfont \\
EOF
cat >> $DESTDIR/x11/share/pkgconfig/sdl_ttf-build.pc << EOF
/x11/bin/showfont \\
/x11/lib/libSDL_ttf-1.2.so.0
EOF

exit

# patch
--- SDL_ttf-2.0.11.orig/configure.in
+++ SDL_ttf-2.0.11/configure.in
@@ -1,5 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(README)
+AC_INIT(SDL_ttf, 2.0.11)
+AC_CONFIG_SRCDIR(README)
+AC_CONFIG_MACRO_DIR(acinclude)
 
 dnl Set various version strings - taken gratefully from the GTk sources
 
@@ -11,8 +13,8 @@
 # if backwards compatibility has been broken,
 # set BINARY_AGE and INTERFACE_AGE to 0.
 
-MAJOR_VERSION=2
-MINOR_VERSION=0
+MAJOR_VERSION=1
+MINOR_VERSION=2
 MICRO_VERSION=11
 INTERFACE_AGE=1
 BINARY_AGE=11
@@ -42,7 +44,7 @@
 AC_CANONICAL_HOST
 
 dnl Setup for automake
-AM_INIT_AUTOMAKE(SDL_ttf, $VERSION)
+AM_INIT_AUTOMAKE(foreign)
 
 dnl Check for tools
 AC_PROG_LIBTOOL
@@ -116,9 +118,12 @@
 AC_PATH_PROG(FREETYPE_CONFIG, freetype-config, no)
 no_freetype=""
 if test "$FREETYPE_CONFIG" = "no" ; then
+    PKG_CHECK_MODULES([FT2], [freetype2], [
+    CFLAGS="$CFLAGS $FT2_CFLAGS"
+    LIBS="$LIBS $FT2_LIBS"],
     AC_MSG_ERROR([
 *** Unable to find FreeType2 library (http://www.freetype.org/)
-])
+]))
 else
     CFLAGS="$CFLAGS `$FREETYPE_CONFIG $freetypeconf_args --cflags`"
     LIBS="$LIBS `$FREETYPE_CONFIG $freetypeconf_args --libs`"
@@ -156,14 +161,11 @@
         ;;
     *)
         MATHLIB="-lm"
-        AC_PATH_X
-        AC_PATH_XTRA
-        if test x$have_x = xyes; then
-            CFLAGS="$CFLAGS $X_CFLAGS"
-            SYS_GL_LIBS="$X_LIBS -lGL"
-        else
+        PKG_CHECK_MODULES([SYS_GL], [gl], [
+            CFLAGS="$CFLAGS $SYS_GL_CFLAGS"
+        ], [
             SYS_GL_LIBS="-lGL"
-        fi
+        ])
         ;;
 esac
 AC_MSG_CHECKING(for OpenGL support)

# source
https://www.libsdl.org/projects/SDL_ttf/release/SDL_ttf-2.0.11.tar.gz
